File: app/src/main/java/com/jann_luellmann/thekenapp/data/model/Drink.java
package com.jann_luellmann.thekenapp.data.model;

import com.jann_luellmann.thekenapp.R;
import com.jann_luellmann.thekenapp.data.util.Editable;

import java.math.BigDecimal;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Drink {

    @PrimaryKey(autoGenerate = true)
    private long id;

    @ColumnInfo(name = "name")
    @Editable(stringId = R.string.name)
    private String name;

    @ColumnInfo(name = "price")
    @Editable(stringId = R.string.price)
    private long price;

    private long eventId;

    @NonNull
    @Override
    public String toString() {
        return name + ": " + priceAsText();
    }

    private String priceAsText() {
        return new BigDecimal(price).movePointLeft(2).toString() + " €";
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/model/Drink.kt
package com.jann_luellmann.thekenapp.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.jann_luellmann.thekenapp.R
import com.jann_luellmann.thekenapp.data.util.Editable
import java.math.BigDecimal

@Entity
class Drink(
    @PrimaryKey(autoGenerate = true)
    var drinkId: Long = 0,

    @ColumnInfo(name = "dname")
    @field:Editable(stringId = R.string.identifier)
    var name: String = "",

    @ColumnInfo(name = "price")
    @field:Editable(stringId = R.string.price)
    var price: Long = 0
) {
    override fun toString(): String {
        return name + ": " + priceAsText()
    }

    private fun priceAsText(): String {
        return BigDecimal(price).movePointLeft(2).toString() + " €"
    }
}